web3/transaction.py

```diff
@@ -0,0 +1,66 @@
+
+
+class Transaction(object):
+    IMMUTABILITY_WARNING = "Transactions are immutable, create a new one instead of modifying this"
+
+    def __init__(self, values):
+        '''
+        @param values a dictionary of transaction values
+        '''
+        self.__dict__ = values
+        try:
+            self._unique_txid()
+        except AttributeError as e:
+            assert False, "Transaction must be uniquely defined. Failed because: %s" % e
+        self._locked = True
+
+    def _unique_txid(self):
+        'currently the hash works as a unique identifier, but someday maybe not'
+        return getattr(self, 'hash')
+
+    def __getitem__(self, key):
+        return getattr(self, key)
+
+    def __setitem__(self, key, val):
+        raise NotImplementedError(Transaction.IMMUTABILITY_WARNING)
+
+    def __delitem__(self, key):
+        raise NotImplementedError(Transaction.IMMUTABILITY_WARNING)
+
+    def __iter__(self):
+        return iter(self.__dict__)
+
+    def todict(self):
+        return dict(self.__dict__)
+
+    def items(self):
+        return self.__dict__.items()
+
+    def iteritems(self):
```

Comment:
Is this still required?  I've been under the impression that for `2.7/3.4/3.5` that merely implementing `.items()` was sufficient.